src/utilities/request.jl

```diff
@@ -1,3 +1,21 @@
+# this is used to allow custom dispatches to `_http_request`
+abstract type AbstractBackend end
+
+"""
+    AWS.HTTPBackend <: AWS.AbstractBackend
+
+An `HTTPBackend` can hold default `http_options::AbstractDict{Symbol,<:Any}`
+to pass to HTTP.jl, which can be overwritten per-request by any `http_options`
+supplied there.
+"""
+struct HTTPBackend{T<:AbstractDict{Symbol,<:Any}} <: AbstractBackend
+    http_options::T
+end
```

Comment:
wasn't sure if in this case it's better to do this or just
```suggestion
struct HTTPBackend <: AbstractBackend
    http_options::AbstractDict{Symbol,<:Any}
end
```
which I think matches the choices made elsewhere more. Leaving the type abstract means we pay for dynamic dispatch when we merge these options with the Request's options-- but the Request has abstractly typed fields anyway so maybe we need to do that regardless. Probably in the context of network operations, the microseconds of dynamic dispatch do not really matter at all... And it should need less compile time to make it abstract.